src/domain/settings/regions/components/region-form/use-store-data.tsx

```diff
@@ -0,0 +1,58 @@
+import { FulfillmentProvider, PaymentProvider, Store } from "@medusajs/medusa"
+import { useAdminStore } from "medusa-react"
+import { useMemo } from "react"
+import { countries } from "../../../../../utils/countries"
+import fulfillmentProvidersMapper from "../../../../../utils/fulfillment-providers.mapper"
+import paymentProvidersMapper from "../../../../../utils/payment-providers-mapper"
+
+export const useStoreData = () => {
+  const { store, isLoading } = useAdminStore()
+
+  const { fulfillment_providers, payment_providers } = store as Store & {
+    fulfillment_providers: FulfillmentProvider[]
+    payment_providers: PaymentProvider[]
+  }
+
+  const paymentProviderOptions = useMemo(() => {
+    if (isLoading) {
+      return []
+    }
+
+    return payment_providers?.map((p) => paymentProvidersMapper(p.id)) || []
+  }, [payment_providers, isLoading])
+
+  const fulfillmentProviderOptions = useMemo(() => {
+    if (isLoading) {
+      return []
+    }
+
+    return (
+      fulfillment_providers?.map((p) => fulfillmentProvidersMapper(p.id)) || []
+    )
+  }, [fulfillment_providers, isLoading])
+
+  const currencyOptions = useMemo(() => {
+    if (isLoading) {
+      return []
+    }
+
+    return (
+      store?.currencies.map((c) => ({
```

Comment:
question: Should we move that as part of the hook? The idea would be to do it only once when the store is loaded and not have to do it everywhere we need that. the hook could return storeFormattedCurrencies?

Reply:
There is no common pattern between different places where we are creating currency options, some use `currency_code.toUpperCase()` as the label others use the currency name. When we have a single approach to rendering currency and country option then we can do it, but until then I think this is fine as the caching makes sure that the request to fetch store is only done once and shared across everywhere that it is used. 😄